### ISSUE TITLE
[Klarna] Support payments in other presentment currencies when allowed

### ISSUE THREAD

**Issue by danielmx-dev** [Automattic/woocommerce-payments#8984]
### Describe the bug
Klarna payments in general are restricted to domestic transactions; however, there is one exception that is outlined in [the current documentation](https://docs.stripe.com/payments/klarna):

> If you’re based in the EEA, UK, or Switzerland, then you can transact with consumers across the EEA, UK, and Switzerland, provided the presentment currency matches the currency of the customer’s country. For example, a Swedish business can present in EUR to accept Klarna from a buyer in Germany.

However, due to existing validations in the code, the Klarna payment method is not available when using a presentment currency different from the domestic currency used in the country that was selected during the WooPayments onboarding process.

### To Reproduce

1. Connect your store to a Stripe account from a Klarna-supported European country that doesn't use Euro as their domestic currency (e.g. United Kingdom)
2. Enable Klarna
3. As a shopper, set the currency to EUR, add a product to your cart and proceed to checkout
4. As the billing country, select one Klarna-supported country that uses Euro as their domestic currency (e.g. Belgium)
5. Notice the list of payment methods

### Actual behavior
Klarna is not available as an option in the Payment Methods list.

#### Checkout
![image](https://github.com/Automattic/woocommerce-payments/assets/15204776/a10813ef-c5b7-41f4-ac6f-4f7bbb74c349)

### Expected behavior
Klarna should be available as an option in the Payment Methods list, and the order should succeed when using it.

### Additional context
This issue related to #8718, however while the linked issue is focused on hiding Klarna when the location conditions are not met, while this issue should take care of showing Klarna when both the location and currency conditions are met.

The validations that are preventing Klarna from being displayed are mainly these two, since they force to only show the payment method when the cart currency matches the WooPayments account currency: 

https://github.com/Automattic/woocommerce-payments/blob/fa7554041d08430a60fa232f9a91864d56bce398/includes/payment-methods/class-upe-payment-method.php#L173-L195

https://github.com/Automattic/woocommerce-payments/blob/fa7554041d08430a60fa232f9a91864d56bce398/includes/payment-methods/class-upe-payment-method.php#L229-L239

One of the biggest problems that would need to be solved is to figure out if the solution can be implemented entirely in the backend, or if it would be required to also implement frontend changes.

**Comment by vbelolapotkov** [comment]
I think the current behavior is by design. Yes, it has limitation, but I don't think it's a bug, so changed the type to enhancement. 
@c-shultz @pierorocca sending this one to you for further assessment and prioritization.

**Comment by pierorocca** [comment]
Yes that's right by design. Launch was scope limited to exclude intra-Europe currency use cases. Enhancement type would be the best fit. Thanks Vasily.

**Comment by AashikP** [comment]
@vbelolapotkov @pierorocca 

As per [Customer currency and BNPL - Documentation](https://woocommerce.com/document/woopayments/payment-methods/buy-now-pay-later/#customer-currency-and-bnpl), Norway should be using NOK currency, but with that currency, you will not see the Klarna Option:

<img width="372" alt="image" src="https://github.com/user-attachments/assets/31abfd1e-5094-4409-bfb1-b8ecaf981beb">

You need to enable USD currency for this to work. Is this expected?

User report in 8494388-zen

**Comment by shohanhossainnab** [comment]
In 9116526-zen, the merchant's account country is NL, but UK customers don't see the Klarna method.

**Comment by zain0ul0abiddin** [comment]
**Another merchant hit this today, plus two UX inconsistencies worth folding into the scope.**

Zendesk 11496778<br>Textbook case for this issue: account country **GB**, store base country **IE**, store currency **EUR**, `capabilities.klarna_payments: active`, Klarna gateway `enabled: true`. Klarna never renders. Confirmed on the live checkout: `wcpayConfig.paymentMethodsConfig` contained only `card`, and the express checkout element was passed `publicOptions[paymentMethods][klarna]=never`. Blocked at `class-wc-payment-gateway-wcpay.php:4640` via `is_currency_valid( get_account_domestic_currency() )`, since `get_account_domestic_currency()` resolves GB to GBP against a EUR cart and Klarna is domestic-only.<br><br>**Two inconsistencies that made this expensive to diagnose:**<br><br>1. **The cart/product BNPL messaging contradicts checkout.** `WC_Payments_Payment_Method_Messaging_Element::init()` (`class-wc-payments-payment-method-messaging-element.php:109-130`) filters BNPL methods only on *enabled + capability active*, and passes the shopper's billing/store country to Stripe. It never applies the account-country rule that checkout applies. On this store the messaging element ran with `countryCode=IE&currency=EUR&paymentMethodTypes[0]=klarna` and rendered a Klarna promo, while checkout offered no Klarna. The merchant reasonably read the cart message as proof Klarna was working, and had already resorted to hiding it with CSS. We are effectively advertising a payment method the same store cannot offer.<br><br>2. **Settings let you enable Klarna with no signal that it can never render.** The capability is `active`, so the "missing requirements / rejected capabilities" messaging added in 9.4.0 does not fire. Nothing in the settings UI indicates the account-country/currency mismatch, so the merchant's configuration looks correct from every screen they can see.<br><br>**Suggestion — either direction resolves it, but the current middle ground is the worst of both:**<br><br>- **Preferred:** support the use case as originally scoped here, i.e. allow EEA/UK/CH accounts to present in another EEA currency per [Stripe's Klarna docs](<https://docs.stripe.com/payments/klarna>).<br>- **Minimum viable, if that stays in Backlog:** make the product self-consistent. Surface an explicit warning in WooPayments settings when a BNPL method is enabled but cannot render for the current account country and store currency, do not silently enable it (or auto-disable it), and apply the same eligibility gate to the cart/product messaging element so it stops promoting a method checkout will not offer.<br><br>The second option is comparatively cheap and would remove the whole class of support ticket even while the enhancement waits. This is at least the third merchant report on this issue (9116526-zen and 8494388-zen are already cited above).

cc: @pierorocca
